sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.createApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -467,6 +467,64 @@ describe("ApplicationOfferingChangeRequestInstitutionsController(e2e)-createAppl
       });
   });
 
+  it("Should throw program is not active error when trying to submit an application offering request for an inactive program.", async () => {
+    // Arrange
+    const savedUser = await db.user.save(createFakeUser());
+    // Student has a completed application to the institution.
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { institutionLocation: collegeFLocation },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Student SIN Validation.
+    application.student.sinValidation = createFakeSINValidation({
+      student: application.student,
+    });
+    application.data.howWillYouBeAttendingTheProgram =
+      OfferingIntensity.fullTime;
+
+    await db.student.save(application.student);
+    await db.application.save(application);
+
+    // New offering with full time intensity.
+    const fakeOffering = createFakeEducationProgramOffering({
+      auditUser: savedUser,
+      isProgramActive: false,
+      institutionLocation: collegeFLocation,
+    });
+    // Updating study period, that belongs to the program year.
+    fakeOffering.studyStartDate = addDays(
+      5,
+      application.programYear.startDate,
+    ).toISOString();
+    fakeOffering.studyEndDate = addDays(
+      85,
+      application.programYear.startDate,
+    ).toISOString();
+    await db.educationProgramOffering.save(fakeOffering);
+
+    const payload = {
+      applicationId: application.id,
+      offeringId: fakeOffering.id,
+      reason: "Test reason.",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.INTERNAL_SERVER_ERROR)
+      .expect({
+        statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+        message:
+          "Error while submitting an application offering change request",
+        error: "Internal Server Error",
+      });
+  });
```

Comment:
We should never expect a internal server error in the code. If Internal server error is manually thrown from implementation, we should change it.